### PR TITLE
Refactor build pipeline & package.json's of SDK-Core & SDK-Redux

### PR DESCRIPTION
**Pull request by kasparkallas**
* Unify tsconfig's and build pipeline between sdk-core and sdk-redux
* Compile typescript build output to a modern version of ecmascript
* Get rid of rollup
* Use browserify for UMD browser build
* Fix out of date packages inside workspace root's package.json which caused issue (i.e. the "babel-eslint" error)

@0xdavinchee Please take over and write an example for the UMD stuff...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,19 +19,17 @@
         "postinstall": "husky install"
     },
     "devDependencies": {
-        "ajv": "^8.6.3",
-        "babel-eslint": "^10.1.0",
```
Comment (kasparkallas) [inline package.json:23]: `"babel-eslint"` is outdated package. `"ajv"` was never meant to be here.

File: package.json [hunk 2]
```diff
@@ -19,19 +19,17 @@
...
-        "eslint": "^7.20.0",
```
Comment (kasparkallas) [inline package.json:24]: `"eslint"` version 7 was causing issues because rest of the stuff is on 8.

File: packages/sdk-core/package.json [hunk 3]
```diff
@@ -9,18 +9,27 @@
...
-        "dist/**/*.js",
-        "dist/**/*.d.ts"
+        "dist/main",
+        "dist/module",
+        "dist/docs",
+        "dist/index.umd.js",
+        "!**/*.json",
+        "CHANGELOG.md",
+        "LICENSE",
+        "README.md"
```
Comment (kasparkallas) [inline packages/sdk-core/package.json:19]: I unified this section with `sdk-redux`. Looks OK, right?

File: packages/sdk-core/package.json [hunk 4]
```diff
@@ -9,18 +9,27 @@
...
     ],
     "scripts": {
-        "build": "yarn generate-abi-files && yarn generate-web3-types && yarn generate-graphql-types && rm -rf dist && rollup --config rollup.config.ts --configPlugin typescript",
+        "build": "yarn generate-abi-files && yarn generate-web3-types && yarn generate-graphql-types && rm -rf dist && run-p build:*",
+        "build:main": "tsc -p tsconfig.json",
+        "build:module": "tsc -p tsconfig.module.json",
+        "build:umd": "browserify src/index.ts -s sdkCore -p [ tsify --noImplicitAny ] > dist/index.umd.js",
```
Comment (kasparkallas) [inline packages/sdk-core/package.json:25]: Browserify does the UMD magic here by creating a global variable names `sdkCore`. Feel free to change the name.

File: packages/sdk-core/package.json [hunk 5]
```diff
@@ -78,5 +89,8 @@
     },
     "resolutions": {
         "graphql": "^16.0.0"
+    },
+    "engines": {
+        "node": ">=14"
```
Comment (kasparkallas) [inline packages/sdk-core/package.json:94]: Node 14 requirement now.

File: packages/sdk-core/tsconfig.json [hunk 6]
```diff
@@ -1,80 +1,97 @@
 {
```
Comment (kasparkallas) [inline packages/sdk-core/tsconfig.json:1]: Bunch of whitespace changes here.

File: packages/sdk-core/tsconfig.module.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+    "extends": "./tsconfig",
+    "compilerOptions": {
+        "target": "ES2020",
+        "outDir": "dist/module",
+        "module": "esnext",
+        "declaration": true,
+        "declarationMap": true,
+        "sourceMap": true
```
Comment (kasparkallas) [inline packages/sdk-core/tsconfig.module.json:9]: Generating source maps and declaration maps now.

Comment (0xdavinchee) [reply packages/sdk-core/tsconfig.module.json:9]: 🔥

File: packages/sdk-core/package.json (hunk 3 above)
Comment (0xdavinchee) [reply packages/sdk-core/package.json:17]: yeah

File: packages/sdk-core/package.json (hunk 4 above)
Comment (0xdavinchee) [reply packages/sdk-core/package.json]: name is good

**Review by 0xdavinchee — APPROVED** [review]
gmi

**Pull request merged**